sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeed.cs

```diff
@@ -263,43 +263,55 @@ internal DataFeedDetailPatch GetPatchModel()
 
         private static void SetAuthenticationProperties(DataFeedDetail detail, DataFeedSource dataSource)
         {
+            string authentication;
+
             switch (dataSource)
             {
                 case AzureBlobDataFeedSource s:
-                    detail.AuthenticationType = s.GetAuthenticationTypeEnum();
+                    authentication = s.Authentication?.ToString();
+                    detail.AuthenticationType = (authentication == null) ? default(AuthenticationTypeEnum?) : new AuthenticationTypeEnum(authentication);
```

Comment:
We have an extensible enum called `AuthenticationTypeEnum`. It's internal and used by the service to differentiate authentication types.

In the public API, on the other hand, we have 4 different `AuthenticationType` enums (one for each supported `DataFeedSource`). This discussion is not directly related to this PR, but we designed it like this to make it easier for users to get authentication right without much effort (hopefully).

Because of this difference, we needed methods to transform from the internal enum to the public ones, and vice-versa (`GetAuthenticationTypeEnum` was responsible for making the conversion public => internal). Since the public enums are extensible enums now, they all share the same inner string, and conversion became way simpler, so we're removing the helper methods.